modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java

```diff
@@ -1423,6 +1440,9 @@ public DefaultCodegen() {
         // option to change the order of form/body parameter
         cliOptions.add(CliOption.newBoolean(CodegenConstants.PREPEND_FORM_OR_BODY_PARAMETERS,
                 CodegenConstants.PREPEND_FORM_OR_BODY_PARAMETERS_DESC).defaultValue(Boolean.FALSE.toString()));
+        cliOptions.add(CliOption.newBoolean(CodegenConstants.DISCRIMINATOR_EXPLICIT_MAPPING_VERBOSE,
```

Comment:
This option needs better definition. If it's a CLI option (in `Generate.java`), it should be hooked up to `CodegenConfigurator` and applied as a way to configure this generation.

`CliOptions` is a place to define generator-specific (`*Codegen` types) logic, and not "global" workflows.

For instance, the option in `Generate.java` will show via `cli help generate` while the option here would show via `cli config-help -g generator_name`. The latter is intended only for those configurations which don't apply to all generators.

A good rule for these:  only define things in CliOptions if they are template customizable, put others via CodegenConfigurator and extend via additionalProperties.  I open to clarify and clean this all up by 5.0.

Reply:
This has been changed to LEGACY_DISCRIMINATOR_BEHAVIOR
I will update to the suggested integration